### PR TITLE
Decomp map_preview_graphic.c

### PR DESCRIPTION
**Pull request by adrienntindall**

### CODE REVIEW

File: src/unk_0206A360.c [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+        unk->unk20 = GetBgPriority(unk->bgConfig, 0);
+        unk->unk24 = GetBgPriority(unk->bgConfig, 1);
+        unk->unk28 = GetBgPriority(unk->bgConfig, 3);
+        SetBgControlParam(unk->bgConfig, 2, GF_BG_CNT_SET_COLOR_MODE, 1);
+        SetBgPriority(3, 0);
+        SetBgPriority(2, 1);
+        SetBgPriority(0, 2);
+        sub_0206A5D0(unk);
+        sub_0206A660(unk);
+        G2x_SetBlendAlpha_((u32) &reg_G2_BLDCNT, 4, 0x21, 16, 0);
```
Comment (tgsm) [inline src/unk_0206A360.c:122]:
```suggestion
        G2_SetBlendAlpha(4, 0x21, 16, 0);
```

File: src/unk_0206A360.c [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+        BeginNormalPaletteFade(0, 1, 1, 0, 0x10, 1, unk->heapId);
+        unk->state++;
+        break;
+    case 1:
+        if (IsPaletteFadeFinished()) {
+            unk->state++;
+        }
+        break;
+    case 2:
+        if (gSystem.newKeys & 1) {
```
Comment (tgsm) [inline src/unk_0206A360.c:132]:
```suggestion
        if (gSystem.newKeys & PAD_BUTTON_A) {
```

File: src/unk_0206A360.c [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+        unk->unk24 = GetBgPriority(unk->bgConfig, 1);
+        unk->unk28 = GetBgPriority(unk->bgConfig, 3);
+        SetBgControlParam(unk->bgConfig, 2, GF_BG_CNT_SET_COLOR_MODE, 1);
+        SetBgPriority(3, 0);
+        SetBgPriority(2, 1);
+        SetBgPriority(0, 2);
+        sub_0206A5D0(unk);
+        sub_0206A660(unk);
+        G2x_SetBlendAlpha_((u32) &reg_G2_BLDCNT, 4, 0x21, 16, 0);
+        BeginNormalPaletteFade(0, 1, 1, 0, 0x10, 1, unk->heapId);
```
Comment (tgsm) [inline src/unk_0206A360.c:123]:
```suggestion
        BeginNormalPaletteFade(0, 1, 1, RGB_BLACK, 16, 1, unk->heapId);
```

File: src/unk_0206A360.c [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+            unk->state++;
+        }
+        break;
+    case 2:
+        if (gSystem.newKeys & 1) {
+            unk->unk18 = 0;
+            unk->state++;
+            break;
+        }
+        if (++unk->unk4 > 0x3c) {
```
Comment (tgsm) [inline src/unk_0206A360.c:137]:
```suggestion
        if (++unk->unk4 > 60) {
```

File: src/unk_0206A360.c [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+        case 0:
+            PlayerAvatar_ToggleAutomaticHeightUpdating_NowApply(fsys->playerAvatar, 0);
+            unkSub->state++;
+            break;
+        case 1:
+            VecFx32 pos;
+            player = PlayerAvatar_GetMapObject(fsys->playerAvatar);
+            unkSub->unk2++;
+            MapObject_GetPositionVec(player, &pos);
+            pos.y -= (1 << 13);
```
Comment (tgsm) [inline src/unk_0206A360.c:200]:
```suggestion
            pos.y -= 2 * FX32_ONE;
```

File: src/unk_0206A360.c [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+    LocalMapObject *player;
+    
+    if (!unk->unk30) {
+        return TRUE;
+    }
+
+    UnkStateMachineSubstruct_0206A388 *unkSub = &unk->unk34; //required to match
+    
+    switch (unk->unk34.state) {
+        case 0:
```
Comment (PikalaxALT) [inline src/unk_0206A360.c:191]: Deindent `case` label

File: src/unk_0206A360.c [hunk 7]
```diff
@@ -0,0 +1,235 @@
...
+
+static void sub_0206A5D0(UnkStruct_0206A388 *unk) {
+    GfGfxLoader_GXLoadPal(NARC_a_1_5_0, _0210FAE8[unk->unk16].unk8[unk->unk2C].palNo, GF_PAL_LOCATION_MAIN_BG, GF_PAL_SLOT_0_OFFSET, 0x160, unk->heapId);
+    GfGfxLoader_LoadCharData(NARC_a_1_5_0, _0210FAE8[unk->unk16].unk8[unk->unk2C].charNo, unk->bgConfig, GF_BG_LYR_MAIN_2, 0, 0, 1, unk->heapId);
+    GfGfxLoader_LoadScrnData(NARC_a_1_5_0, _0210FAE8[unk->unk16].unk8[unk->unk2C].scrnNo, unk->bgConfig, GF_BG_LYR_MAIN_2, 0, 0, 1, unk->heapId);
+}
+
+static void sub_0206A660(UnkStruct_0206A388 *unk) {
+    unk->unk10 = AllocWindows(unk->heapId, 1);
+    AddWindowParameterized(unk->bgConfig, unk->unk10, 2, 0, 0, 0x1C, 2, 0xd, 0x2C0);
```
Comment (PikalaxALT) [inline src/unk_0206A360.c:178]: name unk10; represent coords, dims, and palette num in base 10

File: src/unk_0206A360.c [hunk 8]
```diff
@@ -0,0 +1,235 @@
...
+            break;
+        case 1:
+            VecFx32 pos;
+            player = PlayerAvatar_GetMapObject(fsys->playerAvatar);
+            unkSub->unk2++;
+            MapObject_GetPositionVec(player, &pos);
+            pos.y -= (1 << 13);
+            MapObject_SetPositionVec(player, &pos);
+            if (unkSub->unk2 >= 16) {
+                PlayerAvatar_ToggleAutomaticHeightUpdating_NowApply(fsys->playerAvatar, 1);
```
Comment (PikalaxALT) [inline src/unk_0206A360.c:203]:
```suggestion
                PlayerAvatar_ToggleAutomaticHeightUpdating_NowApply(fsys->playerAvatar, TRUE);
```

File: src/unk_0206A360.c [hunk 9]
```diff
@@ -0,0 +1,235 @@
...
+
+    unk->state = 0;
+    unk->unk4 = 0;
+    unk->bgConfig = FieldSystem_GetBgConfigPtr(fsys);
+    unk->unk18 = 0;
+    unk->unk1A = 0;
+    unk->unk16 = index;
+
+    switch (time) {
+    case 0:
```
Comment (PikalaxALT) [inline src/unk_0206A360.c:78]: Time of day constants

File: src/unk_0206A360.c (hunk 6 above)
Comment (PikalaxALT) [reply src/unk_0206A360.c:191]: Also, state consts

File: asm/unk_0206A360_asm.s [hunk 10]
```diff
@@ -11,543 +11,9 @@
...
-	mov r0, #0
-	add sp, #0xc
-	pop {r3, r4, r5, r6, pc}
-	.balign 4, 0
-	thumb_func_end sub_0206A694
-
 	.data
 
+.public _0210FAE8
 _0210FAE8:
```
Comment (PikalaxALT) [inline asm/unk_0206A360_asm.s:17]: Port to C. Since it's in `.data`, it's non-const, but still static

**Review by PikalaxALT — CHANGES_REQUESTED** [review]

File: asm/unk_0206A360_asm.s (hunk 10 above)
Comment (adrienntindall) [reply asm/unk_0206A360_asm.s:17]: was saving for post dumping the narc, just trying to get it rebuilt at the moment but I got sick lol

File: include/map_preview_graphic.h [hunk 11]
```diff
@@ -0,0 +1,9 @@
+#ifndef POKEHEARTGOLD_MAP_PREVIEW_GRAPHIC_H
+#define POKEHEARTGOLD_MAP_PREVIEW_GRAPHIC_H
+
+#include "task.h"
+
+u8 MapPreviewGraphic_GetIndex(u32 mapId);
+void MapPreviewGraphic_BeginShowImage(TaskManager *man, int index, u8 time, int a3);
+
+#endif //POKEHEARTGOLD_MAP_PREVIEW_GRAPHIC_H
```
Comment (tgsm) [inline include/map_preview_graphic.h:9]: Missing newline at end of file

File: src/map_preview_graphic.c [hunk 12]
```diff
@@ -0,0 +1,946 @@
...
+    case PG_STATE_WAIT_OR_SKIP:
+        if (gSystem.newKeys & PAD_BUTTON_A) {
+            unk->fadeOutTimer = 0;
+            unk->state++;
+            break;
+        }
+        if (++unk->displayTimer > 60) {
+            unk->fadeOutTimer = 0;
+            unk->state++;
+
```
Comment (tgsm) [inline src/map_preview_graphic.c:851]: Extraneous newline

File: src/map_preview_graphic.c [hunk 13]
```diff
@@ -0,0 +1,946 @@
...
+        break;
+    case 2:
+        Field_PlayerAvatar_OrrTransitionFlags(fsys->playerAvatar, 1);
+        Field_PlayerAvatar_ApplyTransitionFlags(fsys->playerAvatar);
+        unkSub->state++;
+        break;
+    case 3:
+        player = PlayerAvatar_GetMapObject(fsys->playerAvatar);
+        if (MapObject_AreBitsSetForMovementScriptInit(player)) {
+            MapObject_SetHeldMovement(player, 1);
```
Comment (tgsm) [inline src/map_preview_graphic.c:926]:
```suggestion
            MapObject_SetHeldMovement(player, MOVEMENT_FACE_DOWN);
```

File: src/map_preview_graphic.c [hunk 14]
```diff
@@ -0,0 +1,946 @@
...
+        MapObject_GetPositionVec(player, &pos);
+        pos.y -= 2 * FX32_ONE;
+        MapObject_SetPositionVec(player, &pos);
+        if (unkSub->unk2 >= 16) {
+            PlayerAvatar_ToggleAutomaticHeightUpdating_NowApply(fsys->playerAvatar, TRUE);
+            unkSub->state++;
+        }
+        break;
+    case 2:
+        Field_PlayerAvatar_OrrTransitionFlags(fsys->playerAvatar, 1);
```
Comment (tgsm) [inline src/map_preview_graphic.c:919]:
```suggestion
        Field_PlayerAvatar_OrrTransitionFlags(fsys->playerAvatar, PLAYER_TRANSITION_WALKING);
```

File: src/map_preview_graphic.c [hunk 15]
```diff
@@ -0,0 +1,946 @@
...
+            unkSub->state++;
+        }
+        break;
+    case 4:
+        player = PlayerAvatar_GetMapObject(fsys->playerAvatar);
+        if (MapObject_AreBitsSetForMovementScriptInit(player)) {
+            if (FollowMon_IsActive(fsys)) {
+                ov01_02205790(fsys, 1);
+                sub_0205FC94(FollowMon_GetMapObject(fsys), 0x30);
+                sub_02069DC8(FollowMon_GetMapObject(fsys), 1);
```
Comment (tgsm) [inline src/map_preview_graphic.c:936]:
```suggestion
                sub_02069DC8(FollowMon_GetMapObject(fsys), TRUE);
```

File: src/map_preview_graphic.c [hunk 16]
```diff
@@ -0,0 +1,946 @@
...
+            MapObject_SetHeldMovement(player, 1);
+            unkSub->state++;
+        }
+        break;
+    case 4:
+        player = PlayerAvatar_GetMapObject(fsys->playerAvatar);
+        if (MapObject_AreBitsSetForMovementScriptInit(player)) {
+            if (FollowMon_IsActive(fsys)) {
+                ov01_02205790(fsys, 1);
+                sub_0205FC94(FollowMon_GetMapObject(fsys), 0x30);
```
Comment (tgsm) [inline src/map_preview_graphic.c:935]: 0x30 is a movement constant; you can add it in, or don't

File: src/map_preview_graphic.c [hunk 17]
```diff
@@ -0,0 +1,946 @@
...
+    
+    if (!unk->unk30) {
+        return TRUE;
+    }
+
+    UnkStateMachineSubstruct_0206A388 *unkSub = &unk->unk34; //required to match
+    
+    switch (unk->unk34.state) {
+    case 0:
+        PlayerAvatar_ToggleAutomaticHeightUpdating_NowApply(fsys->playerAvatar, 0);
```
Comment (tgsm) [inline src/map_preview_graphic.c:903]:
```suggestion
        PlayerAvatar_ToggleAutomaticHeightUpdating_NowApply(fsys->playerAvatar, FALSE);
```

**Review by tgsm — CHANGES_REQUESTED** [review]

File: src/map_preview_graphic.c (hunk 16 above)
Comment (adrienntindall) [reply src/map_preview_graphic.c:935]: Might as well add it in as an unk to be documented later

File: src/map_preview_graphic.c [hunk 18]
```diff
@@ -0,0 +1,946 @@
...
+}
+
+static int sub_0206A694(UnkStruct_0206A388 *unk, FieldSystem *fsys) {
+    LocalMapObject *player;
+    
+    if (!unk->unk30) {
+        return TRUE;
+    }
+
+    UnkStateMachineSubstruct_0206A388 *unkSub = &unk->unk34; //required to match
```
Comment (tgsm) [inline src/map_preview_graphic.c:899]: This comment seems kinda redundant

**Review by tgsm — APPROVED** [review]
I have no major complaints (yet)

File: src/map_preview_graphic.c (hunk 18 above)
Comment (adrienntindall) [reply src/map_preview_graphic.c:899]: it was just a really weird thing to be necessary tbh

File: files/fielddata/graphic/preview_graphic/preview_graphic.mk [hunk 19]
```diff
@@ -0,0 +1,18 @@
+PREVIEW_GRAPHIC_DIR := files/fielddata/graphic/preview_graphic/preview_graphic
+PREVIEW_GRAPHIC_NARC := $(PREVIEW_GRAPHIC_DIR).narc
+
+PREVIEW_GRAPHIC_PNG	 := $(wildcard $(PREVIEW_GRAPHIC_DIR)/*.png)
+PREVIEW_GRAPHIC_NSCR := $(wildcard $(PREVIEW_GRAPHIC_DIR)/*.NSCR)
```
Comment (PikalaxALT) [inline files/fielddata/graphic/preview_graphic/preview_graphic.mk:5]: fix spacing

File: files/fielddata/graphic/preview_graphic/preview_graphic/.gitignore [hunk 20]
```diff
@@ -0,0 +1 @@
+*.NCGR
```
Comment (PikalaxALT) [inline files/fielddata/graphic/preview_graphic/preview_graphic/.gitignore:1]: missing newline at eof

**Review by PikalaxALT — CHANGES_REQUESTED** [review]

**Review by PikalaxALT — APPROVED** [review]

**Pull request merged**